### PR TITLE
MGDOBR-646: Slack Event Sources Integration test

### PR DESCRIPTION
**Pull request by jomarko**
The added test cratues two processors
- Slack Source processor - observing slack channel and recreating new cloud event
- Slack Action/Sink processor - takes cloud event of ^ and resend new message to slack channel

The first slack message is created via slack webhook url
```
    And create message with text "Slack Event Source Feature trigger" on slack channel
```

The new generated message is checked as
```
    Then Slack channel contains message with text "Message Slack Event Source Feature trigger was observed" within 1 minute
```

**Question**
The message on slack should contain some timestamp or other identiefier to be the message always unique? In other words so we are usre we didn't read old message from slack?

[MGDOBR-646](https://issues.redhat.com/browse/MGDOBR-646)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/SlackSteps.java [hunk 1]
```diff
@@ -31,4 +32,15 @@ public void slackActionTest(String messageText, int timeoutMinutes) {
...
     }
+
+    @And("^create message with text \"([^\"]*)\" on slack channel$")
+    public void createMessageOnSlackChannel(String messageText) {
+        String messageTextWithoutPlaceholders = ContextResolver.resolveWithScenarioContext(context, messageText);
+        Awaitility.await()
+                .atMost(Duration.ofMinutes(1))
+                .pollInterval(Duration.ofSeconds(1))
+                .untilAsserted(
+                        () -> assertThat(SlackResource.postToSlackWebhookUrl(messageTextWithoutPlaceholders)).isEqualTo(200));
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/SlackSteps.java:43]: Why is this step implemented as a loop? Does the Slack POST request fail occasionally?

File: integration-tests/src/test/resources/features/source-slack.feature [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            "webhookUrl": "${env.slack.webhook.url}"
+          }
+      },
+      "filters": [
+        {
+          "type": "StringEquals",
+          "key": "type",
+          "value": "SlackSource"
+        }
+      ],
```
Comment (sutaakar) [inline integration-tests/src/test/resources/features/source-slack.feature:51]: Is this needed?

**Comment by sutaakar** [comment]
> The message on slack should contain some timestamp or other identifier to be the message always unique? In other words so we are sure we didn't read old message from slack?

Yes, can you please add a possibility to insert a unique id? I can be for example something like `${uuid.<UUID name>}`.

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/SlackSteps.java (hunk 1 above)
Comment (jomarko) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/SlackSteps.java:43]: I think just a consequence of copy and paste during development, will remove it

File: integration-tests/src/test/resources/features/source-slack.feature (hunk 2 above)
Comment (kostola) [reply integration-tests/src/test/resources/features/source-slack.feature:51]: This filter ensures that only `SlackSource` events are processed, but, if you're sure that no other events will ever be sent to that instance (which can be a good assumption in an isolated test), you can remove this filter with no unwanted effects.

File: integration-tests/src/test/resources/features/source-slack.feature [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            "token": "${env.slack.webhook.token}"
+          }
+      },
+      "filters": [
+        {
+          "key": "data.text",
+          "type": "StringContains",
+          "values": ["Slack Event Source Feature trigger"]
+        }
+      ]
```
Comment (kostola) [inline integration-tests/src/test/resources/features/source-slack.feature:28]: What is the purpose of this filter instead?

Comment (jomarko) [reply integration-tests/src/test/resources/features/source-slack.feature:28]: We uses the same *mc* slack channel for multiple tests. If you check https://app.slack.com/client/T033KAWDGUC/C0346PP8EL8 there are also messages like "hello world ..."

So in my test I want to listen only for specific messages on *mc* slack channel

File: integration-tests/src/test/resources/features/source-slack.feature [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          "type": "StringBeginsWith",
+          "values": ["Slack Event Source Feature trigger"]
+        }
+      ]
+    }
+    """
+    And the Processor "slackSourceProcessor" of the Bridge "mybridge" is existing with status "ready" within 5 minutes
+    And the Processor "slackSourceProcessor" of the Bridge "mybridge" has source of type "Slack"
+
+    When add a Processor to the Bridge "mybridge" with body:
```
Comment (sutaakar) [inline integration-tests/src/test/resources/features/source-slack.feature:34]: Nitpick, if you move creating `slackForwardProcessor` right after creating `slackSourceProcessor` and then you check availability of both, you can save significant amount of test time (as both Processors will start in parallel).

File: integration-tests/README.md [hunk 5]
```diff
@@ -42,10 +42,11 @@ Test identifier is a specific resource identifier used in a scenario (for exampl
 
 It is possible to use various placeholders in feature files, allowing dynamic acess to runtime values and system identifiers. The placeholders use format `${placeholder-value}`.
 
-Curently there is a possibility to use these placeholders in feature files:
+Currently, there is a possibility to use these placeholders in feature files:
 - ${env.`<System property name>`} to use a System property
 - ${bridge.`<Bridge name>`.id} to use actual "Bridge id"
 - ${cloud-event.`<Cloud event id>`.id} to use "System cloud event id" (Cloud event id which is actually used for Cloud event invocation)
+- ${uuid.`<Uuid name>`} TODO
```
[the remark this answers is not in the document]
Comment (jomarko) [reply integration-tests/README.md]: done

File: integration-tests/src/test/resources/features/source-slack.feature (hunk 4 above)
Comment (jomarko) [reply integration-tests/src/test/resources/features/source-slack.feature:31]: done

**Review by manstis — APPROVED** [review]
In the absence of anybody else; the test reads OK although I'm no Cucumber expert.

:+1:

**Review by jiripetrlik — APPROVED** [review]
LGTM

**Pull request merged**